tools/dissector/sample_dissector.cpp

```diff
@@ -31,13 +32,171 @@ namespace OpenDDS
   namespace DCPS
   {
 
+    bool utf16_to_utf8(
+      std::string & to, ACE_CDR::WChar* from, size_t length = 1
+    ) {
+      size_t size = length * sizeof(ACE_CDR::WChar);
+      gchar* from_;
+
+#if ACE_SIZEOF_WCHAR == 4
+      /*
+       * We have to trim the extra 16 bits of zeros between the characters
+       * before we can pass it to glib's g_convert. Otherwise it will
+       * interpret them as NULL characters and stop after the first two bytes.
+       * g_convert doesn't respsect the size passed to it if it thinks it can
+       * keep going.
+       */
+      uint16_t* trimmed = new uint16_t[length + 1];
```

Comment:
Could this be std::vector instead?  That would reduce complexity of dealing with dynamic allocation here.